### PR TITLE
ODAA-201: Delete stuff

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/ODAA-201

Adds check for safe deletion of entities and helpers to delete entities (can be improved and generalized in the future).
Adds anonymization of users.

### CODE REVIEW

File: src/ReferenceManager/DataFlowReferenceManager.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                        ]
+                    ));
+                }
+            }
+        }
+
+        return $messages;
+    }
+
+    public function delete($dataSource, bool $flush = true)
```
Comment (lats1) [inline src/ReferenceManager/DataFlowReferenceManager.php]: $dataSource should be $dataFlow maybe?

File: src/ReferenceManager/DataFlowReferenceManager.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        $this->translator = $translator;
+    }
+
+    public function supports($entity): bool
+    {
+        return $entity instanceof DataFlow;
+    }
+
+    /**
+     * Get a list of messages telling why a given user cannot be deleted.
```
Comment (lats1) [inline src/ReferenceManager/DataFlowReferenceManager.php]: user in comment should probably be data flow

File: src/ReferenceManager/DataSourceReferenceManager.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        $this->translator = $translator;
+    }
+
+    public function supports($entity): bool
+    {
+        return $entity instanceof DataSource;
+    }
+
+    /**
+     * Get a list of messages telling why a given user cannot be deleted.
```
Comment (lats1) [inline src/ReferenceManager/DataSourceReferenceManager.php]: user in comment should probably be data source

File: src/ReferenceManager/UserReferenceManager.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+use FOS\UserBundle\Model\UserManagerInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class UserReferenceManager implements ReferenceManagerInterface
+{
+    /** @var UserManagerInterface */
+    private $userManager;
+
+    /** @var DataFlowRepository */
```
Comment (lats1) [inline src/ReferenceManager/UserReferenceManager.php]: Shouldn't this be EntityManagerInterface?

File: src/ReferenceManager/UserReferenceManager.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    {
+        return $entity instanceof User;
+    }
+
+    /**
+     * Get a list of messages telling why a given user cannot be deleted.
+     *
+     * @return Message[]
+     */
+    public function getDeleteMessages($dataFlow): array
```
Comment (lats1) [inline src/ReferenceManager/UserReferenceManager.php]: $dataFlow should probably be $user

File: templates/data_flow/_table.html.twig [hunk 6]
```diff
@@ -36,6 +36,7 @@
                         <div class="btn-group" role="group" aria-label="">
                             <a href="{{ path('data_flow_edit', {'id': data_flow.id}) }}" class="btn text-primary m-0 p-0 mr-3" data-toggle="tooltip" title="{{ 'Edit flow'|trans }}"><i class="fal {{ iconClass('edit') }} mr-1"></i><span class="sr-only">{{ 'Edit flow'|trans }}</span></a>
                             <a href="{{ path('data_flow_transforms_index', {'data_flow': data_flow.id}) }}" class="btn text-primary m-0 p-0 mr-3 mr-2" data-toggle="tooltip" title="{{ 'Change recipe'|trans }}"><i class="fal {{ iconClass('recipe') }} mr-1"></i><span class="sr-only">{{ 'Change recipe'|trans }}</span></a>
+                            <a href="{{ path('data_flow_delete', {'id': data_flow.id}) }}" class="btn text-danger m-0 p-0 mr-3 mr-2" data-toggle="tooltip" title="{{ 'Delete flow'|trans }}"><i class="fal {{ iconClass('delete') }} mr-1"></i><span class="sr-only">{{ 'Delete recipe'|trans }}</span></a>
```
Comment (lats1) [inline templates/data_flow/_table.html.twig]: Should 'Delete recipe' be 'Delete flow'?

**Review by lats1 — CHANGES_REQUESTED** [review]

File: src/ReferenceManager/DataFlowReferenceManager.php (hunk 1 above)
Comment (rimi-itk) [reply src/ReferenceManager/DataFlowReferenceManager.php]: Yes.

File: src/ReferenceManager/DataFlowReferenceManager.php (hunk 2 above)
Comment (rimi-itk) [reply src/ReferenceManager/DataFlowReferenceManager.php]: Yes.

File: src/ReferenceManager/DataSourceReferenceManager.php (hunk 3 above)
Comment (rimi-itk) [reply src/ReferenceManager/DataSourceReferenceManager.php]: Yes.

File: src/ReferenceManager/UserReferenceManager.php (hunk 4 above)
Comment (rimi-itk) [reply src/ReferenceManager/UserReferenceManager.php]: Yes.

File: src/ReferenceManager/UserReferenceManager.php (hunk 5 above)
Comment (rimi-itk) [reply src/ReferenceManager/UserReferenceManager.php]: Yes.

File: templates/data_flow/_table.html.twig (hunk 6 above)
Comment (rimi-itk) [reply templates/data_flow/_table.html.twig]: Yes.

File: src/ReferenceManager/DataFlowReferenceManager.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+use App\Entity\DataFlow;
+use App\Entity\DataTransform;
+use App\Repository\DataFlowRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class DataFlowReferenceManager implements ReferenceManagerInterface
+{
+    /** @var DataFlowRepository */
```
Comment (lats1) [inline src/ReferenceManager/DataFlowReferenceManager.php]: Should be EntityManagerInterface

File: src/ReferenceManager/DataSourceReferenceManager.php [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+use App\Entity\DataFlow;
+use App\Entity\DataSource;
+use App\Repository\DataFlowRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class DataSourceReferenceManager implements ReferenceManagerInterface
+{
+    /** @var DataFlowRepository */
```
Comment (lats1) [inline src/ReferenceManager/DataSourceReferenceManager.php]: Should be EntityManagerInterface

**Review by lats1 — CHANGES_REQUESTED** [review]

**Review by lats1 — APPROVED** [review]

**Pull request merged**